auth/allow_all_authenticator.hh

```diff
@@ -75,6 +75,14 @@ public:
         return make_ready_future<custom_options>();
     }
 
+    virtual bool uses_salted_hashes() const override {
+        return false;
+    }
+
+    virtual future<std::optional<sstring>> query_salted_hash(std::string_view role_name) const override {
```

Comment:
I have some reservation to the name of those functions, since it's interface is should describe it on a higher level (because what even is salted hash for the reader here?), maybe call it:

get_password_hash(), whether it's salted or it does query it's impl. detail
similarly uses_salted_hashes -> can_get_password_hash ?